### ISSUE TITLE
Troubles with tag_index in cassandra

### ISSUE THREAD

**Issue by heruv1m** [jaegertracing/jaeger#3028]
I add attribute with [api](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/Span.java#L97) and export spans via [JaegerGrpcSpanExporter](https://github.com/open-telemetry/opentelemetry-java/blob/50408d499f85d5761d0a5ed9bf9d77d5ff01fff5/exporters/jaeger/src/main/java/io/opentelemetry/exporter/jaeger/JaegerGrpcSpanExporter.java)
We are using cassandra as storage backend. 
Finally I have that attributes as tags in table **tag_index**. It affects cassandra's permormance drammaticaly.
 An example of warning from cassandra

> large_data - Writing a partition with too many rows [opentelemetry/tag_index:myApp:status.code:0] (207365 rows)

What is the right way to put additional attributes with jaeger backend?
Is there any way to disable that index?

**Comment by yurishkuro** [comment]
You can exclude certain tags from being indexed via `--cassandra.index.tag-blacklist`

**Comment by heruv1m** [comment]
Thank you for your answer, I disabled it as you recomended. 
But, I stille have troubles with another tables.
Is it really to many data for one service? Can you give any advice how it can be fixed?

> WARN  2021-05-27 14:24:10,430 [shard 1] large_data - Writing a partition with too many rows [opentelemetry/duration_index:myAppName:/myAppName/full:2021-05-27T11:00:00] (266047 rows)
WARN  2021-05-27 14:26:47,639 [shard 2] large_data - Writing a partition with too many rows [opentelemetry/service_operation_index:myAppName:/myAppName/myFirstEndpoint] (372596 rows)
WARN  2021-05-27 14:28:40,689 [shard 1] large_data - Writing a partition with too many rows [opentelemetry/service_operation_index:myAppName:/session/secondEndpoint] (446916 rows)
WARN  2021-05-27 14:29:47,294 [shard 2] large_data - Writing a partition with too many rows [opentelemetry/duration_index:myAppName:/user_notes/resolving_tokens:2021-05-27T11:00:00] (498160 rows)
WARN  2021-05-27 14:31:21,547 [shard 2] large_data - Writing a partition with too many rows [opentelemetry/service_name_index:myAppName:3] (154147 rows)
WARN  2021-05-27 14:31:22,074 [shard 2] large_data - Writing a partition with too many rows [opentelemetry/service_name_index:myAppName:1] (154040 rows)
WARN  2021-05-27 14:31:22,600 [shard 2] large_data - Writing a partition with too many rows [opentelemetry/service_name_index:myAppName:4] (153858 rows)
WARN  2021-05-27 14:32:14,217 [shard 1] large_data - Writing a partition with too many rows [opentelemetry/service_name_index:myAppName:8] (154244 rows)
WARN  2021-05-27 14:32:14,720 [shard 1] large_data - Writing a partition with too many rows [opentelemetry/service_name_index:myAppName:0] (153826 rows)
WARN  2021-05-27 14:32:15,169 [shard 1] large_data - Writing a partition with too many rows [opentelemetry/service_name_index:myAppName:9] (153671 rows)

**Comment by yurishkuro** [comment]
TL;DR: this is why we recommend Elasticsearch.

Maybe you can bump the corresponding settings in Cassandra, I don't recall seeing these warnings for just 100,000 entries. Beyond that there's not much that can be done. There is an internal sharding/bucketing done by jaeger, it's possible to increase the number of buckets (although it might be hard coded and will need code changes to expose the param), but it will make the reads slower since at read time there is a fan out to all buckets.

**Comment by heruv1m** [comment]
Thanks for answer. We switched to Elasticsearch, this is no longer relevant
